### ISSUE TITLE
Can't run your example:  Error 400 bad request

### ISSUE THREAD

**Issue by byteptr** [panodata/grafana-pandas-datasource#7]
Hello, 

I tried to run `grafana_example.py`, then I successfully added SimpleJSON datasource, but I can't get this working.

When I try to configure the datasource to `127.0.0.1:3003` this gives me

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>400 Bad Request</title>
<h1>Bad Request</h1>
<p>Failed to decode JSON object: Expecting value: line 1 column 1 (char 0)</p>
```

Here is my config:
![grsck](https://user-images.githubusercontent.com/35638772/146845762-9aa6d3e1-4f36-4c36-aa0a-0de235a27fb6.png)

When I search on the internet it seems to be a Flask error, but I'm not sure. Can you say me if I'm doing something wrong please? How can I get sinewave data ? 

Thanks

**Comment by amotl** [comment]
Dear Raul,

thanks for your report. We will need to gather more information to get some idea what could have gone wrong on your end.

- Can you outline the steps you have been taking in more detail? Did you follow any specific documentation hosted within this repository? If so, which was it?
- Can you see any exception on the console where you started the Python service?
- Last but not least, may I also humbly ask you about the Grafana version you are using? As far as I can remember, this has been conceived when Grafana 7 was the latest and greatest, so there might be incompatibilities with Grafana 8.

With kind regards,
Andreas.

**Comment by amotl** [comment]
Dear Raul,

the documentation at [^1] and [^2] has been significantly improved. Maybe this helps. Please let us know about the outcome for you.

Please also note:
> When running Grafana in Docker, use `host.docker.internal` to address the Docker host [where the Flask HTTP service is running].

With kind regards,
Andreas.

[^1]: https://github.com/panodata/grafana-pandas-datasource/blob/main/README.rst
[^2]: https://github.com/panodata/grafana-pandas-datasource/blob/main/examples/sinewave-midnights/readme.rst

**Comment by amotl** [comment]
Dear Raul,

finally, we have been able to discover the root cause of this problem at https://github.com/panodata/grafana-pandas-datasource/issues/9#issuecomment-1029537267 and provided a fix with #10. Please find corresponding details about this problem in the section below.

Thank you again for your report! Please let us know if you are observing any other issues. We will also be happy to hear back if it works for you now.

With kind regards,
Andreas.

## Background
The problem happens because the Simple JSON Datasource might send upstream HTTP requests with empty request bodies, but still with the `Content-Type: application/json` HTTP header present. Some web frameworks do not handle that well and will attempt to decode the empty string from JSON, which will just croak. That can easily be reproduced like
```
$ python3 -c 'import json; json.loads("")'
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

In this case, it happens with the Flask web framework. We are currently using Flask version 1.1.4. Please note that nothing violates the HTTP specification here. It is a) perfectly valid, though unusual, that GET requests would carry HTTP request bodies and it is b) also perfectly valid, that a zero-length request body will be submitted. Then, if the web framework will attempt to decode that request body from JSON, without applying any other sanitations, it will obviously croak.

## Discussions
- https://github.com/pallets/flask/issues/1317
- https://github.com/pallets/flask/issues/1614
- https://github.com/pallets/flask/pull/1940
- https://github.com/flask-restful/flask-restful/issues/50
- https://github.com/flask-restful/flask-restful/issues/387
- https://github.com/flask-restful/flask-restful/issues/510
- https://github.com/flask-restful/flask-restful/issues/556

## Workarounds
- `StripContentTypeMiddleware` by @vrde: https://github.com/bigchaindb/bigchaindb/commit/e272a4e0d
  Via: https://github.com/flask-restful/flask-restful/issues/510#issuecomment-312845061

## Fixes
- https://github.com/flask-restful/flask-restful/pull/56
- https://github.com/flask-restful/flask-restful/pull/366

## Solution

@ThiefMaster added at https://github.com/pallets/flask/pull/1940#issuecomment-229936524:

> You can catch the exception or use `request.get_json(silent=True)`.

This is what has been implemented with #10. Thank you!

**Comment by amotl** [comment]
Hi again,

`grafana-pandas-datasource-0.3.0` has been released and includes this bugfix.

With kind regards,
Andreas.
